core/model/src/main/java/org/eclipse/rdf4j/model/util/Statements.java

```diff
@@ -121,4 +124,92 @@ public static boolean isSameTriple(Statement st1, Statement st2) {
 		return st1.getPredicate().equals(st2.getPredicate()) && st1.getSubject().equals(st2.getSubject())
 				&& st1.getObject().equals(st2.getObject());
 	}
+
+	/**
+	 * Converts the supplied RDF* statement to RDF reification statements, and sends the resultant statements to the
+	 * supplied consumer. If the supplied statement is not RDF* it will be sent to the consumer as is.
+	 * <p>
+	 * The statements needed to represent reification will use blank nodes.
+	 *
+	 * @param st       the {@link Statement} to convert.
+	 * @param consumer the {@link Consumer} function for the produced statements.
+	 */
+	public static void convertRDFStarToReification(Statement st, Consumer<Statement> consumer) {
+		convertRDFStarToReification(SimpleValueFactory.getInstance(), st, consumer);
+	}
+
+	/**
+	 * Converts the supplied RDF* statement to RDF reification statements, and sends the resultant statements to the
+	 * supplied consumer. If the supplied statement is not RDF* it will be sent to the consumer as is.
+	 * <p>
+	 * The statements needed to represent reification will use blank nodes.
+	 * <p>
+	 * The supplied value factory is used to create all new statements and blank nodes.
+	 *
+	 * @param vf       the {@link ValueFactory} to use for creating statements.
+	 * @param st       the {@link Statement} to convert.
+	 * @param consumer the {@link Consumer} function for the produced statements.
+	 */
+	public static void convertRDFStarToReification(ValueFactory vf, Statement st, Consumer<Statement> consumer) {
+		convertRDFStarToReification(vf, (t) -> vf.createBNode(), st, consumer);
```

Comment:
You are right. As I started from Statement I didn't really consider this in the context of e.g. Model. Even though the converted data will be OK (you can reach the subject/predicate/object of the rdf:Statement) I agree it would be best to have a consistent mapping for identical triples. I just committed a fix that uses the triple's stringValue() + base64 encoding to derive the BNode ID. It uses SimpleValueFactory (before it used the passed value factory) to ensure that you get the ID you want. I also extended the tests to cover this case.